rmw_opendds_cpp/src/rmw_client.cpp

```diff
@@ -12,259 +12,158 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <string>
-
-#include "rmw/allocators.h"
-#include "rmw/error_handling.h"
-#include "rmw/impl/cpp/macros.hpp"
-#include "rmw/rmw.h"
-
-#include "rmw_opendds_cpp/types.hpp"
-#include "rmw_opendds_cpp/qos.hpp"
-
-#include "rmw_opendds_cpp/opendds_static_client_info.hpp"
-#include "rmw_opendds_cpp/identifier.hpp"
-#include "process_topic_and_service_names.hpp"
-#include "type_support_common.hpp"
+#include <rmw_opendds_cpp/DDSClient.hpp>
+#include <rmw_opendds_cpp/OpenDDSNode.hpp>
+#include <rmw_opendds_cpp/identifier.hpp>
+#include <rmw_opendds_cpp/qos.hpp>
+#include <rmw_opendds_cpp/types.hpp>
 
+#include <dds/DCPS/DomainParticipantImpl.h>
 #include <dds/DCPS/Marked_Default_Qos.h>
 
 // Uncomment this to get extra console output about discovery.
-// This affects code in this file, but there is a similar variable in:
-//   rmw_opendds_cpp/shared_functions.cpp
 // #define DISCOVERY_DEBUG_LOGGING 1
 
-#include <dds/DCPS/DomainParticipantImpl.h>
+#include <rmw/allocators.h>
```

Comment:
These rmw includes should be below the rmw_opendds_cpp but above the dds ones.

Reply:
Yea, we have a card created for that issue: https://opendds.atlassian.net/browse/RMW-91. For now, since most header files are not in the same directory with the cpp files, the double-quotes are replaced with angle-brackets. The order is basically from the narrowest scope to widest scope. The final rmw include rules are still needed to be defined when we come to RMW-91. (For example, when we use OpenDDS code to implement rmw functions, should OpenDDS code be considered a narrower scope? Also, for class files, the class header should be the first one included. What about the function implementation files, should we put rmw headers or our implementation headers first?)